lib/rmg/CMapGenerator.cpp

```diff
@@ -62,10 +62,89 @@ CMapGenerator::CMapGenerator(CMapGenOptions& mapGenOptions, int RandomSeed) :
 	zonesTotal(0), tiles(nullptr), prisonsRemaining(0),
     monolithIndex(0)
 {
+	loadConfig();
 	rand.setSeed(this->randomSeed);
 	mapGenOptions.finalize(rand);
 }
 
+void CMapGenerator::loadConfig()
+{
+	static std::map<std::string, ETerrainType> terrainMap
+	{
+		{"dirt", ETerrainType::DIRT},
+		{"sand", ETerrainType::SAND},
+		{"grass", ETerrainType::GRASS},
+		{"snow", ETerrainType::SNOW},
+		{"swamp", ETerrainType::SWAMP},
+		{"subterranean", ETerrainType::SUBTERRANEAN},
+		{"lava", ETerrainType::LAVA},
+		{"rough", ETerrainType::ROUGH}
+	};
+	static const std::map<std::string, Res::ERes> resMap
+	{
+		{"wood", Res::ERes::WOOD},
+		{"ore", Res::ERes::ORE},
+		{"gems", Res::ERes::GEMS},
+		{"crystal", Res::ERes::CRYSTAL},
+		{"mercury", Res::ERes::MERCURY},
+		{"sulfur", Res::ERes::SULFUR},
+		{"gold", Res::ERes::GOLD},
+	};
+	static std::map<std::string, ERoadType::ERoadType> roadTypeMap
+	{
+		{"dirt_road", ERoadType::DIRT_ROAD},
+		{"gravel_road", ERoadType::GRAVEL_ROAD},
+		{"cobblestone_road", ERoadType::COBBLESTONE_ROAD}
+	};
+	static const ResourceID path("config/randomMap.json");
+	JsonNode randomMapJson(path);
+	for(auto& s : randomMapJson["terrain"]["undergroundAllow"].Vector())
+	{
+		if(!s.isNull())
+			config.terrainUndergroundAllowed.push_back(terrainMap[s.String()]);
```

Comment:
A bit of code duplication. Small one so and it will be hard to get rid of it probably. Only extending JsonNode with something like VectorOfInts() VectorOfStrings() might save a few lines of code.